app/file.go

```diff
@@ -71,14 +72,26 @@ func ReadFile(path string) ([]byte, *model.AppError) {
 		accessKey := utils.Cfg.FileSettings.AmazonS3AccessKeyId
 		secretKey := utils.Cfg.FileSettings.AmazonS3SecretAccessKey
 		secure := *utils.Cfg.FileSettings.AmazonS3SSL
+		encrypted := *utils.Cfg.FileSettings.EnableClientSideEncryption
 		signV2 := *utils.Cfg.FileSettings.AmazonS3SignV2
 		s3Clnt, err := s3New(endpoint, accessKey, secretKey, secure, signV2)
 		if err != nil {
 			return nil, model.NewLocAppError("ReadFile", "api.file.read_file.s3.app_error", nil, err.Error())
 		}
 		bucket := utils.Cfg.FileSettings.AmazonS3Bucket
-		minioObject, err := s3Clnt.GetObject(bucket, path)
-		defer minioObject.Close()
```

Comment:
@coreyhulen  my lack of Go experience is a problem here.  The minio GetEncryptedObject returns an io.Reader object, which doesn't have a close method.  The GetObject method that was used before returns an "Object" instead.

And, as I look more closely here, I'm not really sure what type should be declared on line 83 as I don't know what common type I can use for both of those methods.  Any ideas?